src/app.js

```diff
@@ -130,13 +130,22 @@ app.routeAsync("/charon/getAvailable")
   .getAsync(endpoints.charon.getAvailable);
 
 app.routeAsync("/charon/getDataset")
-  .getAsync(endpoints.charon.getDataset);
+  .getAsync(
+    endpoints.charon.setSourceFromPrefix,
+    endpoints.charon.setDatasetFromPrefix,
+    endpoints.charon.canonicalizeDatasetPrefix,
+    endpoints.charon.getDataset);
```

Comment:
It's nice to see how simpler things are after the prefix munging is split out into separate middleware.

Reply:
Yeah! Much of the complexity is from that proto-router.